packages/labs/forms/src/form-associated.ts

```diff
@@ -0,0 +1,456 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {type ReactiveElement} from '@lit/reactive-element';
+
+// #region Types
+
+export interface FormValueOptions<T> {
+  converter?: FormValueConverter<T>;
+}
+
+interface FormValueConverter<T> {
+  toFormValue(v: T): FormValue;
+  fromFormValue(v: FormValue): T;
+}
+
+export type FormValue = string | File | FormData | null;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Constructor<T = object> = new (...args: any[]) => T;
+
+export interface FormAssociated extends ReactiveElement {
+  /**
+   * Implement this method to validate the element. Return a `ValidityResult`.
+   */
+  _getValidity?(): ValidityResult;
```

Comment:
Code style question; Has this pattern (public properties/methods starting with underscore) been used previously? If not, would `protected` with a declared class be an option?

Reply:
Another option is using symbols, which has a few benefits in terms of "hiding" the method from outsiders and making it explicit for implementers:

```typescript
export const getValidity = Symbol("getValidity");

export interface FormAssociated extends ReactiveElement {
    [getValidity]?(): ValidityResult;
}
```